rpcserver.go

```diff
@@ -1810,10 +1810,19 @@ func (s *rpcServer) prepareLeasesResponse(ctx context.Context,
 					bidDuration = match.Order.(*order.Bid).LeaseDuration
 				}
 
+				// The clearing price is dependent on the lease
+				// duration, except for older batch versions
+				// where there was just one single duration.
+				duration := bidDuration
+				if batch.Version != order.DefaultVersion {
```

Comment:
I don't understand why this is needed. Won't the bid duration always match the duration for the batch it was matched in?

Reply:
This is for legacy legacy orders where the durations were defined free style. When de-serializing those batch snapshots, the single clearing price is put into the map with the 2016 duration as the key. Otherwise we wouldn't know whether to look it up using the bid or ask duration (as they could differ before the uniform 2016 duration was introduced).